### PR TITLE
Create Process Story Service

### PR DESCRIPTION
**Pull request by andrewyguo**
## Notion ticket link

[Ticket ](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=ba22ac93d09040ff8219c60ac7caabdb&p=bc4ee7a18fb04effbb86a4e9da0bd362)

## Implementation description

- Create process story service that reads file and extracts story information without committing changes to the database

## Steps to test

1. Place this file in your root directory 
[test_story_file.docx](https://github.com/uwblueprint/planet-read/files/7803441/test_story_file.docx)

2. [Log in as Angela Merkel on GQL](http://localhost:5000/graphql?query=%0Amutation%20%7B%0A%20%20login%20(email%3A%22planetread%2Bangelamerkel%40uwblueprint.org%22%2C%20password%3A%22%22)%20%7B%0A%20%20%20%20accessToken%0A%20%20%20%20refreshToken%0A%20%20%20%20id%0A%20%20%7D%0A%7D&operationName=undefined) and retrieve the auth_token, set the variable `auth_token`

3. Run `curl http://localhost:5000/graphql -F operations='{"query": "mutation ($file: Upload!) { processStory(storyFile: $file) { storyContents }}","variables": {"file": null}}' -F map='{ "0": ["variables.file"]}' -F 0=@test_story_file.docx -H "Authorization: Bearer $auth_token"`
4. This command should return the contents of the file

## What should reviewers focus on?

- Correct information is being returned from calling the mutation
- 

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/APIClients/mutations/StoryMutations.ts [hunk 1]
```diff
@@ -199,4 +199,17 @@ export const SOFT_DELETE_STORY = gql`
     }
   }
 `;
+
 export type SoftDeleteStoryResponse = { ok: boolean };
+
+// export const PROCESS_STORY = gql`
```
Comment (andrewyguo) [inline frontend/src/APIClients/mutations/StoryMutations.ts:205]: Added this thinking that the ticket involved frontend changes as well. I think it's a good idea to keep this for the next person who works on the frontend side of this ticket

File: backend/python/migrations/versions/6bcd31d976d6_create_languages_table.py [hunk 2]
```diff
@@ -5,8 +5,8 @@
 Create Date: 2021-12-31 22:23:00.847824
 
 """
-from alembic import op
 import sqlalchemy as sa
+from alembic import op
```
Comment (andrewyguo) [inline backend/python/migrations/versions/6bcd31d976d6_create_languages_table.py:9]: The linter keeps on changing this :( 

`docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."
All done! ✨ 🍰 ✨
88 files left unchanged.
Fixing /app/migrations/versions/6bcd31d976d6_create_languages_table.py`

File: backend/python/app/services/interfaces/story_service.py [hunk 3]
```diff
@@ -249,3 +249,18 @@ def get_languages(self):
         Get a list of languages currently used in the platform
         """
         pass
+
+    @abstractmethod
+    def import_story(self, details, file):
+        """
+        Import story from word doc
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:256]: nit: Import story from word doc and save as Story & StoryContents in db

File: frontend/src/APIClients/mutations/StoryMutations.ts [hunk 4]
```diff
@@ -199,4 +199,20 @@ export const SOFT_DELETE_STORY = gql`
     }
```
Comment (gaoxk) [inline frontend/src/APIClients/mutations/StoryMutations.ts:199]: can you revert the changes in this file? uploading files via frontend likely wont work like this (jenn L is working on this for uploading resumes) 
generally also should avoid merging stuff we cant/havent tested

**Review by gaoxk — COMMENTED** [review]
gotta meeting coming up, will test pr afterwards
edit: oops meant to hit "request changes"

File: backend/python/app/graphql/mutations/story_mutation.py [hunk 5]
```diff
@@ -295,3 +296,22 @@ def mutate(root, info, story_details, story_file):
         except Exception as e:
             error_message = getattr(e, "message", None)
             raise Exception(error_message if error_message else str(e))
+
+
+class ProcessStory(graphene.Mutation):
+    class Arguments:
+        story_file = Upload(required=True)
+
+    story_contents = graphene.Field(lambda: graphene.List(StoryContentsResponseDTO))
```
Comment (gaoxk) [inline backend/python/app/graphql/mutations/story_mutation.py:305]: i think its fine if the return type is just `graphene.Field(lambda: graphene.List(graphene.String))`, the 
id, story_id, line_index, is_deleted isnt necessary

File: backend/python/app/services/implementations/story_service.py [hunk 6]
```diff
@@ -96,6 +96,27 @@ def import_story(self, details, file):
             self.logger.error(str(error))
             raise error
 
+    def process_story(self, story_file):
+        try:
+            story_contents = self._read_doc(story_file["path"])
+
+            new_contents = [
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:103]: ditto comment here

**Review by gaoxk — CHANGES_REQUESTED** [review]
tested and worked as expected

File: backend/python/app/graphql/mutations/story_mutation.py (hunk 5 above)
Comment (andrewyguo) [reply backend/python/app/graphql/mutations/story_mutation.py:305]: Wait don't we need the line index when we display the story preview?

File: backend/python/app/graphql/mutations/story_mutation.py [hunk 7]
```diff
@@ -10,6 +10,7 @@
 from ..types.story_type import (
     CreateStoryTranslationRequestDTO,
     CreateStoryTranslationResponseDTO,
+    StoryContentsResponseDTO,
```
Comment (gaoxk) [inline backend/python/app/graphql/mutations/story_mutation.py:13]: nit: this is an unused import, remove

File: backend/python/app/services/implementations/story_service.py [hunk 8]
```diff
@@ -96,6 +96,16 @@ def import_story(self, details, file):
             self.logger.error(str(error))
             raise error
 
+    def process_story(self, story_file):
+        try:
+            story_contents = self._read_doc(story_file["path"])
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:101]: the only thing this function does is call `_read_doc`, can we instead just rename `_read_doc->process_story` and pass in the correct path fromo the process_story gql query?

File: frontend/src/APIClients/mutations/StoryMutations.ts [hunk 9]
```diff
@@ -199,4 +199,5 @@ export const SOFT_DELETE_STORY = gql`
...
   }
 `;
+
```
Comment (gaoxk) [inline frontend/src/APIClients/mutations/StoryMutations.ts:202]: nit: remove new line

**Review by gaoxk — APPROVED** [review]
feel free to merge after making changes. fyi theres no auth requirement on the mutation rn so this is the actual call to test:
```
curl http://localhost:5000/graphql -F operations='{"query": "mutation ($file: Upload!) { processStory(storyFile: $file) { storyContents }}","variables": {"file": null}}' -F map='{ "0": ["variables.file"]}' -F 0=@the_musical_donkey.docx 
```

File: frontend/src/APIClients/mutations/StoryMutations.ts (hunk 9 above)
Comment (andrewyguo) [reply frontend/src/APIClients/mutations/StoryMutations.ts:202]: think the linter added this
